xbmc/utils/StringUtils.h

```diff
@@ -141,8 +141,8 @@ class StringUtils
   static bool StartsWithNoCase(const char *s1, const char *s2);
   static bool EndsWith(const std::string &str1, const std::string &str2);
   static bool EndsWith(const std::string &str1, const char *s2);
-  static bool EndsWithNoCase(const std::string &str1, const std::string &str2);
-  static bool EndsWithNoCase(const std::string &str1, const char *s2);
+  static bool EndsWithNoCase(const std::string& str1, const std::string& str2);
+  static bool EndsWithNoCase(const std::string& str1, const char* s2);
```

Comment:
Why do you only fix these formatting issues, not all/none still contained in this header? Change seems completely random.

Reply:
That's an unintended/unavoidable byproduct of trying to include the choice of 2 algorithms in the PR, one of which needs StringUtils changes, and the other one doesn't (and reverts the prev. StringUtils changes - but reverting to original generates formatting warnings).
If algo 1 / commit 2 is chosen, then the .h and .cpp format changes will match.
If the algorithm 2 / commit 3 is chosen, I'll remove the StringUtils changes+revert and the header won't change at all.